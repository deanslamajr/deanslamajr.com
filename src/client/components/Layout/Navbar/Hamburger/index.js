import React from 'react'
import cssModules from 'react-css-modules'
import classNames from 'classnames'

import styles from './hamburger.css'

class Hamburger extends React.Component {
  render () {
    const { onMenuClick, expanded } = this.props

    const isActive = classNames({
      active: expanded
    }, 'mobile', 'hamburger')

    return (
      <div onClick={onMenuClick} styleName={isActive}>
        <span styleName='line' />
        <span styleName='line' />
        <span styleName='line' />
      </div>
    )
  }
}

export default cssModules(Hamburger, styles, { allowMultiple: true })
