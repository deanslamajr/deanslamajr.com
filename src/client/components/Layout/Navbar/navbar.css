@import '../../breakpoints.css';

@import url('https://fonts.googleapis.com/css?family=Josefin+Sans');

.outer-container {
  display: flex;
  justify-content: space-between;
  font-family: 'Josefin Sans', sans-serif;
  font-size: 2.25rem;
  background-color: #004741;
  color: white;
}

.mobile {
  display: none;
}

.middle-container {
  display: flex;
  justify-content: space-between;
  padding: .4rem .2rem .1rem;
}

.inner-container {
  display: flex;
  justify-content: flex-end;
}

.link {
  margin-left: 2rem;
  padding: .4rem .2rem .1rem;
}

.link a {
  text-decoration: none;
  color: white;
}

.active {
  text-decoration: underline;
}

@media (--phone-max-width) {
  .outer-container {
    flex-direction: column;
  }

  .inner-container {
    display: none;
    flex-direction: column;
  }

  .mobile {
    display: inline;
  }

  .link {
    display: block;
  }

  .show {
    display: inline;
  }
}
