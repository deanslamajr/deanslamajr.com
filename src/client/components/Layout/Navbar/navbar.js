import React from 'react';
import classNames from 'classnames';
import { Link } from 'react-router';
import cssModules from 'react-css-modules';

import styles from './navbar.css';

class NavBar extends React.Component {
  _generateItemClasses(items) {
    const { activeItem } = this.props;

    const itemSelected = {};

    items.forEach(item => {
      itemSelected[item] = classNames('link', { 'active': activeItem === item.toLowerCase() });
    });

    return itemSelected;
  }

  _renderMenuItem(item, styleName) {
    const { onLinkClick } = this.props;

    function _onLinkClick(item) {
      onLinkClick(item);
    }

    return (
      <span key={item} onClick={_onLinkClick.bind(this, item)} styleName={styleName}>
        <Link to={`/${item.toLowerCase()}`}>{item}</Link>
      </span>
    );
  }

  render() {
    const {
      title,
      menuItems,
      expanded,
      onMenuClick
    } = this.props;

    const conditionalShow = classNames({
      [styles['show']]: expanded
    });

    const itemsClasses = this._generateItemClasses(menuItems);
    const navbarItems = menuItems.map(item => this._renderMenuItem(item, itemsClasses[item]))

    return (
      <div styleName='outer-container shadow'>
        <span styleName='middle-container'>
          <span>{title}</span>
          <span onClick={onMenuClick} styleName='mobile'>Menu</span>
        </span>
        <span styleName='inner-container' className={conditionalShow}>
          { navbarItems }
        </span>
      </div>
    );
  }
}

NavBar.propTypes = {
  title: React.PropTypes.string.isRequired,
  menuItems: React.PropTypes.arrayOf(React.PropTypes.string),
  expanded: React.PropTypes.bool.isRequired,
  onMenuClick: React.PropTypes.func.isRequired,
  onLinkClick: React.PropTypes.func.isRequired
}

export default cssModules(NavBar, styles, { allowMultiple: true });