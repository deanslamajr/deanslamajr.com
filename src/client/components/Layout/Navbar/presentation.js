import React from 'react';
import classNames from 'classnames';
import { Link } from 'react-router';
import cssModules from 'react-css-modules';

import styles from './navbar.css';

class NavBarPresentation extends React.Component {
  generateItemClasses(items) {
    const { activeItem } = this.props;

    const itemSelected = {};

    items.forEach(item => {
      itemSelected[item] = classNames('link', { 'active': activeItem === item.toLowerCase() });
    });

    return itemSelected;
  }

  renderMenuItem(item, styleName) {
    const { onLinkClick } = this.props;

    function _onLinkClick(item) {
      onLinkClick(item);
    }

    return (
      <span key={item} onClick={_onLinkClick.bind(this, item)} styleName={styleName}>
        <Link to={`/${item.toLowerCase()}`}>{item}</Link>
      </span>
    );
  }

  render() {
    const {
      menuItems,
      expanded,
      onMenuClick
    } = this.props;

    const conditionalShow = classNames({
      [styles['show']]: expanded
    });

    const itemsClasses = this.generateItemClasses(menuItems);
    const navbarItems = menuItems.map(item => this.renderMenuItem(item, itemsClasses[item]))

    return (
      <div styleName='outer-container'>
        <span styleName='middle-container'>
          <span>Dean Slama Jr.</span>
          <span onClick={onMenuClick} styleName='mobile'>Menu</span>
        </span>
        <span styleName='inner-container' className={conditionalShow}>
          { navbarItems }
        </span>
      </div>
    );
  }
}

NavBarPresentation.propTypes = {
  menuItems: React.PropTypes.arrayOf(React.PropTypes.string),
  expanded: React.PropTypes.bool.isRequired,
  onMenuClick: React.PropTypes.func.isRequired,
  onLinkClick: React.PropTypes.func.isRequired
}

export default cssModules(NavBarPresentation, styles, { allowMultiple: true });