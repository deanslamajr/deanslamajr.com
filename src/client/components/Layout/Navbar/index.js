import React from 'react';
import { browserHistory } from 'react-router';

import Navbar from './navbar';

export default class NavBarContainer extends React.Component {
  constructor(props) {
    super(props);

    this._onMenuClick = this._onMenuClick.bind(this);
    this._onLinkClick = this._onLinkClick.bind(this);

    this.state = {
      expanded: false
    };
  }

  _onMenuClick() {
    this.setState({ expanded: !this.state.expanded });
  }

  _onLinkClick(item) {
    const path = item.toLowerCase();

    this.setState({ 
      expanded: false,
      activeItem: path
    });

    browserHistory.push(`/${path}`);
  }

  componentWillMount() {
    // set the activeItem i.e. active view
    const { currentPath, menuItems } = this.props;

    let activeItem = currentPath.split('/')[1];

    activeItem = activeItem
      ? activeItem.toLowerCase()
      : menuItems[0].toLowerCase();

    this.setState({ activeItem });
  }

  render() {
    const { title, menuItems } = this.props;
    return (
      <Navbar
        title={title}
        activeItem={this.state.activeItem}
        expanded={this.state.expanded}
        onMenuClick={this._onMenuClick}
        onLinkClick={this._onLinkClick}
        menuItems={menuItems}
      />
    );
  }
};

NavBarContainer.propTypes = {
  currentPath: React.PropTypes.string.isRequired,
  title: React.PropTypes.string.isRequired,
  menuItems: React.PropTypes.arrayOf(React.PropTypes.string)
}