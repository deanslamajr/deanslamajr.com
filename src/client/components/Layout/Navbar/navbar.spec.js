import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Navbar from '.';

test('Navbar renders correctly', () => {
  const tree = renderer.create(
    <Navbar path={'somePath'}></Navbar>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

test('Navbar expands/collapses on click', () => {
  const navbar = shallow(
    <Navbar path={'somePath'}></Navbar>
  );

  expect(navbar.find('.show').length).toBe(0);

  navbar.find('.mobile').simulate('click');
  expect(navbar.find('.show').length).toBe(1);

  navbar.find('.mobile').simulate('click');
  expect(navbar.find('.show').length).toBe(0);
});