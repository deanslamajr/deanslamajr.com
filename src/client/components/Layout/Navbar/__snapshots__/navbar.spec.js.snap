// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar container should render 1`] = `
<div
  className="outer-container shadow"
>
  <span
    className="middle-container"
  >
    <span>
      test title
    </span>
    <div
      className="mobile hamburger"
      onClick={[Function]}
    >
      <span
        className="line"
      />
      <span
        className="line"
      />
      <span
        className="line"
      />
    </div>
  </span>
  <span
    className="inner-container"
  >
    <span
      className="link"
      onClick={[Function]}
    >
      first
    </span>
    <span
      className="link"
      onClick={[Function]}
    >
      second
    </span>
    <span
      className="link"
      onClick={[Function]}
    >
      third
    </span>
  </span>
</div>
`;

exports[`Navbar should render 1`] = `
<div
  className="outer-container shadow"
>
  <span
    className="middle-container"
  >
    <span>
      test title
    </span>
    <div
      className="mobile hamburger"
      onClick={[Function]}
    >
      <span
        className="line"
      />
      <span
        className="line"
      />
      <span
        className="line"
      />
    </div>
  </span>
  <span
    className="inner-container"
  >
    <span
      className="link"
      onClick={[Function]}
    >
      first
    </span>
    <span
      className="link"
      onClick={[Function]}
    >
      second
    </span>
    <span
      className="link"
      onClick={[Function]}
    >
      third
    </span>
  </span>
</div>
`;
